### DISCUSSION TITLE
VSCode extension : Preview not refreshing

### DISCUSSION THREAD

**Discussion by VivaldoMendes** [quarto-dev/quarto-cli#10745] [Q&A] [answered]
### Description

I am editing a website using Quarto. Everything works fine except for previewing the changes I produce in the document.

Suppose I produce some changes in the document and try previewing them by typing the usual shortcut `Shift+Ctrl+K`. I can see those changes **_only if_** I close the Quarto preview pane and do `Shift+Ctrl+K` again. This is annoying if we have a lot of changes to test in the document. Do you know how I can solve this problem? I searched forums like these and tried to implement the solutions that have been proposed, but I have yet to be successful. I remember a time in the past when this problem did not occur.

I raised this issue next to [#6440](https://github.com/quarto-dev/quarto-cli/issues/6440), but I was advised to create a new one because that issue had already been closed.

I am on a Windows 10 machine, Quarto v1.5.56, and VSCode v1.92.2 (user setup). Apparently, the settings in my VSCode extension are correct. Any help would be very much appreciated because it is terribly annoying to close the Quarto preview pane (and call it back again) to preview any changes I implement in the Quarto file. Thanks.

![a](https://github.com/user-attachments/assets/053ef207-5d38-4385-b1e0-646dcc4e628c)

**Comment by mcanouil** [comment, marked as answer]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Or the detailed steps to reproduce?

The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

Have you consider using `render-on-save`? https://quarto.org/docs/tools/vscode.html#render-on-save

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by attdona** [comment]
I'm on a ubuntu machine and I hit the same problem.

After a little bit of investigation I've found that the issue come from this line that add ""--no-watch-inputs" option to the
preview command, preventing the live update:

https://github.com/quarto-dev/quarto/blob/main/apps/vscode/src/providers/preview/preview.ts#L481

 It seems to me an issue to fix ...

greetings, Attilio

**Reply by mcanouil** [reply]
See my reply about using the "render on save". Preview requires rendering and as rendering can be quite intensive, the "render on save" is disabled by default.
- https://github.com/quarto-dev/quarto-cli/discussions/10745#discussioncomment-10606175

**Comment by padpadpadpad** [comment]
Is this solved now as still happening on my VSCode extension.

**Reply by mcanouil** [reply]
See my reply about using the "render on save". Preview requires rendering and as rendering can be quite intensive, the "render on save" is disabled by default.
- https://github.com/quarto-dev/quarto-cli/discussions/10745#discussioncomment-10606175

**Reply by padpadpadpad** [reply]
Amazing sorry I did not see this!

**Comment by 3xau1o** [comment]
<img width="979" height="504" alt="image" src="https://github.com/user-attachments/assets/fa0517f4-9774-47a5-b358-4d85b7c7c747" />

**Reply by mcanouil** [reply]
@3xau1o thanks but the discussion is quite old and link to documentation about the option was already shared.
Additionally, a comment with a screenshot with no alt-text and no text description does not help future readers.

**Comment by VivaldoMendes** [comment]
Sorry to say but this is terribly annoying. If I use Quarto to produce slides, when I preview the changes I produce in the set of slides, no problem: the preview works after I click on the preview button. In the same machine (a Windows 11), if I update a webpage, it does not work: I have to close the current preview pane, and click once again on preview.

I remember that sometime, in the past, this situation did not occur.

**Reply by mcanouil** [reply]
The team is well aware of the issues with `quarto preview`.
